ray-operator/controllers/ray/common/pod.go

```diff
@@ -691,12 +682,12 @@ func setContainerEnvVars(pod *corev1.Pod, rayNodeType rayv1.RayNodeType, gcsOpti
 	}
 
 	if rayNodeType == rayv1.HeadNode && gcsOptions != nil {
-		container.Env = utils.UpsertEnvVar(container.Env, corev1.EnvVar{
```

Comment:
If `GcsFaultToleranceOptions` is set, `RAY_REDIS_ADDRESS` and `REDIS_PASSWORD` env variables should not be set. Therefore, remove `UpsertEnvVar`.